src/lib.rs

```diff
@@ -4221,6 +4270,161 @@ fn map_darwin_target_from_rust_to_compiler_architecture(target: &Target) -> &str
     }
 }
 
+// Rust and clang/cc don't agree on what equivalent flags should look like either.
+fn rustc_to_cc_flag(flag: &str, family: ToolFamily) -> Option<OsString> {
```

Comment:
Nit: I think there might be value in splitting this into two steps; a "parsing" step, and a . Something like:

```rust
struct CodegenFlags {
    code_model: &'static str,
    vectorize_loops: bool,
    // ...
}

impl CodegenFlags {
    fn parse(rustc_flags: &str) -> CodegenFlags {
        // ...
    }
    
    fn cc_flags(&self, family: ToolFamily) -> Vec<OsString> {
        // ...
    }
}
```

This would make it easier to ensure that we parse the flags equally regardless of the ToolFamily, and might make it easier to combine this information with stuff from `rustc`.

Reply:
good idea, sure that works :)

Reply:
At a second glance, all fields of this struct would need to be Options because we need to distinguish between e.g. true/false/unset for a given flag. It's probably not a huge deal though, should I still do it?

Reply:
Yeah I think that's ok